### PR TITLE
Fix piecewise

### PR DESCRIPTION
**Pull request by paulstapor**
Add support for logical operations in the boolean argument of "sympy.piecewise()"

### CODE REVIEW

File: python/amici/sbml_import.py [hunk 1]
```diff
@@ -1188,41 +1192,53 @@ def _check_unsupported_functions(sym, expression_type, full_sym=None):
...
+        Arguments:
+            math_str: str with mathematical expression
+
+        Returns:
+            math_str: parsed math_str
+
+        Raises:
+    """
+    if ' xor(' in math_str or  'Xor(' in math_str:
+        raise SBMLException('Logical statements including Xor() are '
```
Comment (dweindl) [inline python/amici/sbml_import.py]: Could have added as a single line...

File: python/amici/sbml_import.py [hunk 2]
```diff
@@ -1188,41 +1192,53 @@ def _check_unsupported_functions(sym, expression_type, full_sym=None):
...
 
         Raises:
     """
-    args = tuple(_parse_special_functions(arg, False) for arg in sym._args)
+    args = tuple(_parse_special_functions(arg) for arg in sym._args)
 
+    # Do we have piecewise expressions?
     if sym.__class__.__name__ == 'abs':
         return sp.Abs(sym._args[0])
-    elif sym.__class__.__name__ == 'xor':
```
Comment (dweindl) [inline python/amici/sbml_import.py]: ... here

Comment (dweindl) [reply python/amici/sbml_import.py]: I'd propose to just add the exception line in a new commit on (an up-to-date) `test_action`. That will be the least painfull :)

**Comment by paulstapor** [comment]
As mentioned: This one is branched from `test_action`, instead of develop, as discussed in #771. The actual changes I made should be only in sbml_import.py. Please review all other changes carefully with respect to that.

General question: Does this actually now resolve #771, or rather not?
In any case, it should at the same time fix the problem of having "and"s or "or"s in the piecewise function...

**Comment by dweindl** [comment]
> Does this actually now resolve #771, or rather not?

Yes it does. But maybe you can add a new issue for re-enabling Xor once that is resolved in sympy (referencing your report there).

**Comment by paulstapor** [comment]
> > Does this actually now resolve #771, or rather not?
> 
> Yes it does. But maybe you can add a new issue for re-enabling Xor once that is resolved in sympy (referencing your report there).

On which branch did you implement it? I branched away from test_action around noon today...

**Comment by dweindl** [comment]
> On which branch did you implement it? I branched away from test_action around noon today...

If this refers to 
> What happened to my boolean-handling code?

Then here, and earlier:
https://github.com/ICB-DCM/AMICI/pull/789/commits/e618a6ad9d60c81be4fcd6ae88ca108bd8767198

**Comment by paulstapor** [comment]
Damn, I know what probably happened. I will restore it...

**Comment by paulstapor** [comment]
Now all should be fine again... Sorry for temporary messing up you previous code changes.

**Pull request merged**